tests/bdd/features/api_key.feature

```diff
@@ -0,0 +1,18 @@
+Feature: Api Key
+
+  Scenario: A configured api key is sent in the Authorization header
+    Given an agent
+    When an api key is set to 'RTNxMjlXNEJt' in the config
+    Then the Authorization header is 'ApiKey RTNxMjlXNEJt'
+
+  Scenario: An configured api key takes precedence over a secret token
+    Given an agent
+    When an api key is set in the config
+    When a secret_token is set in the config
+    Then the api key is sent in the Authorization header
+
+  Scenario: A configured secret token is sent if no api key is configured
+    Given an agent
+    When a secret_token is set in the config
+    When an api key is not set in the config
+    Then the secret token is sent in the Authorization header
```

Comment:
Do we have a plan for how to keep these up to date? Not very DRY (and has potential for defeating the purpose of gherkin specs if we accidentally diverge at all from the central version).

Reply:
added download script in 9bfc402